### PR TITLE
CLM-28041: Make initContainer sh command configurable

### PR DESCRIPTION
**Pull request by johnatan-velez**
Jira: https://sonatype.atlassian.net/browse/CLM-28041

Included a new property in Values.yaml so that users can customize the initContainer command.

### CODE REVIEW

File: chart/templates/iq-server-deployment.yaml [hunk 1]
```diff
@@ -218,7 +218,15 @@ spec:
       initContainers:
         - name: {{ .Release.Name }}-set-iq-persistence-ownership
           image: busybox:1.28
-          command: ["sh", "-c", "chown -R 1000:1000 {{ .Values.iq_server.config.clusterDirectory }}"]
+          command:
+            - "sh"
+            - "-c"
+            - >
+              {{- if .Values.customInitCommand -}}
```
Comment (kakumara) [inline chart/templates/iq-server-deployment.yaml:225]: need an description for this in https://github.com/sonatype/nexus-iq-server-ha/blob/main/chart/README.md#chart-configuration-options

File: chart/templates/iq-server-deployment.yaml [hunk 2]
```diff
@@ -218,7 +218,15 @@ spec:
...
+              {{- .Values.customInitCommand -}}
```
Comment (kakumara) [inline chart/templates/iq-server-deployment.yaml:226]: suggestion: `customInitCommand`  --> `pvOwenershipOverride` 
(This is only to customize the chown command, not for anything else)

**Review by juanmartinvidal — APPROVED** [review]

File: chart/templates/iq-server-deployment.yaml (hunk 2 above)
Comment (johnatan-velez) [reply chart/templates/iq-server-deployment.yaml:226]: Maybe it is better to have this under the `iq_server` property? 79f04f3

File: chart/templates/iq-server-deployment.yaml (hunk 1 above)
Comment (johnatan-velez) [reply chart/templates/iq-server-deployment.yaml:225]: Commit: 55959be
Just to clarify, I could say that the default value (default column in the table) is the `chown -R 1000:1000 {{ .Values.iq_server.config.clusterDirectory }}` however it is not really the default value of the `pvOwnershipOverride` property unless I put the default command as value for it in the Values.yaml. What do you think?

File: chart/templates/iq-server-deployment.yaml [hunk 3]
```diff
@@ -218,7 +218,14 @@ spec:
...
         - name: {{ .Release.Name }}-set-iq-persistence-ownership
           image: busybox:1.28
-          command: ["sh", "-c", "chown -R 1000:1000 {{ .Values.iq_server.config.clusterDirectory }}"]
+          command:
+            - "sh"
+            - "-c"
+            - {{- if .Values.iq_server.pvOwnershipOverride }}
+              {{ .Values.iq_server.pvOwnershipOverride }}
+              {{- else }}
+              chown -R 1000:1000 {{ .Values.iq_server.config.clusterDirectory }}
```
Comment (agustin-mrtz) [inline chart/templates/iq-server-deployment.yaml:227]: this doesn't need to be wrapped in quotation marks like it was before?

File: chart/templates/iq-server-deployment.yaml [hunk 4]
```diff
@@ -218,7 +218,14 @@ spec:
       initContainers:
         - name: {{ .Release.Name }}-set-iq-persistence-ownership
           image: busybox:1.28
-          command: ["sh", "-c", "chown -R 1000:1000 {{ .Values.iq_server.config.clusterDirectory }}"]
+          command:
+            - "sh"
```
Comment (agustin-mrtz) [inline chart/templates/iq-server-deployment.yaml:222]: for consistency purposes lets follow the existing syntax like so https://github.com/sonatype/nexus-iq-server-ha/blob/0ba4ddcb6cd0b43dd1162e614a6b5bd02c902347/chart/templates/iq-server-deployment.yaml#L164
```
              command:
                - /bin/sh
                - -c
                - |
                  << helm templating >>
```

**Review by agustin-mrtz — APPROVED** [review]
+1 minor comments

File: chart/templates/iq-server-deployment.yaml (hunk 3 above)
Comment (johnatan-velez) [reply chart/templates/iq-server-deployment.yaml]: using the common sintaxis, it is not necessary.

File: chart/templates/iq-server-deployment.yaml (hunk 1 above)
Comment (kakumara) [reply chart/templates/iq-server-deployment.yaml]: > Commit: [55959be](https://github.com/sonatype/nexus-iq-server-ha/commit/55959be5d05ddd1910789158fa7568ec253a9820)
> Just to clarify, I could say that the default value (default column in the table) is the `chown -R 1000:1000 {{ .Values.iq_server.config.clusterDirectory }}` however it is not really the default value of the `pvOwnershipOverride` property unless I put the default command as value for it in the `Values.yaml`. What do you think?

Let's put that as the default value in Values.yaml

>need an description for this in https://github.com/sonatype/nexus-iq-server-ha/blob/main/chart/README.md#chart-configuration-options
Also, I take my previous comment back. Let's not document this and keep it as an undocumented feature for now.

**Review by kakumara — APPROVED** [review]
+1, with comment https://github.com/sonatype/nexus-iq-server-ha/pull/21#discussion_r1407551752

**Pull request merged**